pkg/controller/node/cidr_set_test.go

```diff
@@ -26,124 +26,167 @@ import (
 )
 
 func TestCIDRSetFullyAllocated(t *testing.T) {
-	_, clusterCIDR, _ := net.ParseCIDR("127.123.234.0/30")
-	a := newCIDRSet(clusterCIDR, 30)
-
-	p, err := a.allocateNext()
-	if err != nil {
-		t.Fatalf("unexpected error: %v", err)
-	}
-	if p.String() != "127.123.234.0/30" {
-		t.Fatalf("unexpected allocated cidr: %s", p.String())
-	}
-
-	_, err = a.allocateNext()
-	if err == nil {
-		t.Fatalf("expected error because of fully-allocated range")
-	}
-
-	a.release(p)
-	p, err = a.allocateNext()
-	if err != nil {
-		t.Fatalf("unexpected error: %v", err)
-	}
-	if p.String() != "127.123.234.0/30" {
-		t.Fatalf("unexpected allocated cidr: %s", p.String())
-	}
-	_, err = a.allocateNext()
-	if err == nil {
-		t.Fatalf("expected error because of fully-allocated range")
+	cases := []struct {
+		clusterCIDRStr string
+		subNetMaskSize int
+		expectedCIDR   string
+	}{
+		{
+			clusterCIDRStr: "127.123.234.0/30",
+			subNetMaskSize: 30,
+			expectedCIDR:   "127.123.234.0/30",
+		},
+		{
+			clusterCIDRStr: "beef:1234::/30",
+			subNetMaskSize: 30,
+			expectedCIDR:   "beef:1234::/30",
+		},
 	}
-}
+	for _, tc := range cases {
+		_, clusterCIDR, _ := net.ParseCIDR(tc.clusterCIDRStr)
+		a := newCIDRSet(clusterCIDR, tc.subNetMaskSize)
```

Comment:
OK. I guess I missed this... so the pool that we allocate from is created at the beginning of the test, so we don't have to worry about cleanup on failures?